addon/utils.js

```diff
@@ -195,3 +195,30 @@ export function parseVariables (valueObj, queryJSON, startPath = '') {
 export function populateQuery (valueObj, query, startPath = '') {
   return JSON.parse(parseVariables(valueObj, JSON.stringify(query), startPath))
 }
+
+/**
+ * Checks if the query object has valid values
+ * @param {Object} value - form value
+ * @param {Object} queryDef - query definition
+ * @param {String} startPath - starting path
+ * @returns {Boolean} true if valid
+ */
+export function hasValidQueryValues (value, queryDef, startPath) {
+  if (!queryDef) {
+    return true
+  }
+
+  const query = populateQuery(value, queryDef, startPath)
+  return Object.keys(query).every((key) => {
+    const prop = query[key]
+
+    // query contains multiple key-value comma-delimited pairs
+    if (prop && (prop.indexOf(':') >= 0 || prop.indexOf(',') >= 0)) {
+      return prop.split(',').every((q) => {
+        return !_.isEmpty(q.split(':')[1])
+      })
+    }
```

Comment:
The `throw` would be much better instead of handling cases where `populateQuery` gave us garbage.  I should have cleaned this up more.